### PR TITLE
feat: add example for spring producer

### PR DESCRIPTION
**Pull request by rkpattnaik780**

### CODE REVIEW

File: kafka-producer-spring/src/main/java/com/example/kafkademo/KafkaConfig.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        var kafkaClientSecret = System.getenv("RHOAS_SERVICE_ACCOUNT_CLIENT_SECRET");
+
+        config.put("bootstrap.servers", System.getenv("KAFKA_HOST"));
+
+        config.put("security.protocol", "SASL_SSL");
+        config.put("sasl.mechanism", "OAUTHBEARER");
+
+        config.put("sasl.jaas.config", "org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required clientId=\"" + kafkaClientId + "\" clientSecret=\"" + kafkaClientSecret + "\" oauth.token.endpoint.uri=\"" + System.getenv("RHOAS_SERVICE_ACCOUNT_OAUTH_TOKEN_URL") + "\";");
+        config.put("sasl.login.callback.handler.class", "org.apache.kafka.common.security.oauthbearer.secured.OAuthBearerLoginCallbackHandler");
+        config.put("sasl.oauthbearer.token.endpoint.url", System.getenv("RHOAS_SERVICE_ACCOUNT_OAUTH_TOKEN_URL"));
```
Comment (andreaTP) [inline kafka-producer-spring/src/main/java/com/example/kafkademo/KafkaConfig.java]: can we group this env var read with the rest?

File: kafka-producer-spring/src/test/java/com/example/kafkademo/KafkaDemoApplicationTests.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.example.kafkademo;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class KafkaDemoApplicationTests {
```
Comment (andreaTP) [inline kafka-producer-spring/src/test/java/com/example/kafkademo/KafkaDemoApplicationTests.java]: let's remove this file if unused

File: kafka-producer-spring/src/main/java/com/example/kafkademo/KafkaProducerExample.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+import java.util.Map;
+
+@SpringBootApplication
+public class KafkaProducerExample implements CommandLineRunner {
+
+	public static void main(String[] args) throws Exception {
+		SpringApplication.run(KafkaProducerExample.class, args);
+
+		SpringApplication app = new SpringApplication(KafkaProducerExample.class);
```
Comment (DuncanDoyle) [inline kafka-producer-spring/src/main/java/com/example/kafkademo/KafkaProducerExample.java]: Any reason for calling the "run" method statically on the class (line 21) and after that instantiating an object and calling the run method again?

Comment (rkpattnaik780) [reply kafka-producer-spring/src/main/java/com/example/kafkademo/KafkaProducerExample.java]: This was a bad copy-paste, thanks for the catch!

**Pull request merged**